packages/ui-components/src/components/pagination/index.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import clsx from 'clsx';
+import { Icon } from '../icons';
+import { SelectBox, SelectBoxOption } from '../select-box';
+import { PaginationProps } from './types';
+import { Input } from '../input/input';
+
+export const Pagination = ({
+  rowsPerPage = [5, 10, 15, 20],
+  currentPage = 1,
+  currentRowsPerPage = 5,
+  totalPage = 1,
+  onChangeRows,
+  onChangePage,
+}: PaginationProps) => {
+  const [page, setPage] = React.useState(currentPage);
+  const [prevStyle, setPrevStyle] = React.useState('');
+  const [nextStyle, setNextStyle] = React.useState('');
+
+  const prevPage = () => {
+    const prevPage = Number(page) > 1 ? Number(page) - 1 : 1;
+    const isFirstPage = prevPage === page;
+    if (!isFirstPage) {
+      setPage(prevPage);
+      onChangePage(prevPage);
+      // animation on button click
+      setPrevStyle(`bg-neutral-bg shadow-[0_0_1px_1px_#ddd]`);
+      setTimeout(() => {
+        setPrevStyle(`bg-transparent `);
+      }, 200);
+    }
+  };
+  const nextPage = () => {
+    const nextPage = Number(page) < totalPage ? Number(page) + 1 : totalPage;
+    const isLastPage = nextPage === page;
+    if (!isLastPage) {
+      setPage(nextPage);
+      onChangePage(nextPage);
+      // animation on button click
+      setNextStyle(`bg-neutral-bg shadow-[0_0_1px_1px_#ddd]`);
+      setTimeout(() => {
+        setNextStyle(`bg-transparent`);
+      }, 200);
+    }
+  };
+  const handleOnchangePage = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const pageNumber = Number(event?.target?.value);
+
+    setPage(
+      Number.isNaN(pageNumber) || pageNumber === 0
```

Comment:
The component and this function should be truthy, does need to check isNan or isNumber, or parsing number, ...